core/module.go

```diff
@@ -177,15 +176,23 @@ func (mod *Module) FromConfig(
 	}
 
 	// Reposition the root of the sourceDir in case it's pointing to a subdir of current sourceDir
-	if cfg.Root != "" {
-		rootPath := filepath.Join("/", filepath.Dir(configPath), cfg.Root)
-		if rootPath != "/" {
-			var err error
+	if filepath.Clean(cfg.Root) != "." {
+		rootPath := filepath.Join(filepath.Dir(configPath), cfg.Root)
+		if rootPath != filepath.Dir(configPath) {
+			configPath, err = filepath.Rel(rootPath, configPath)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get config relative to root: %w", err)
+			}
+			if strings.HasPrefix(configPath+"/", "../") {
```

Comment:
```suggestion
			if strings.HasPrefix(configPath, "../") {
```

Right? This doesn't actually matter in terms of the result, I am just mostly double-checking for clarity. The `configPath` points to a `dagger.json` and we just need to check that the path from root->dagger.json doesn't require a `../` to start.